include/directx/d3d12.h

```diff
@@ -466,8 +466,8 @@ typedef interface ID3D12GraphicsCommandList7 ID3D12GraphicsCommandList7;
 
 
 /* header files for imported files */
-#include "OAIdl.h"
-#include "OCIdl.h"
+#include "oaidl.h"

```

Comment:
Note this one is a little messy...

It has been ``OAIdl.h, ObjIdl.h, ObjIdlbase.h, OCIdl.h, Ole2.h, OleAuto.h, OleCtl.h`` for most of the history of the Windows SDK. For the Windows 10 SDK (20348) they were all made lower-case for 'consistency'.

Reply:
Hm... I didn't realize we did a rename in the Windows SDK. Is there any way to embed the rename into the IDL? Or are we going to have to end up with custom post-processing on the header after building the IDL to get those `#if` to work correctly?

Reply:
MM, the host is Linux, but targeting Windows, so this  won't working

```
[37/1064] Compiling C++ object subprojects/DirectX-Headers-1.0/libDirectX-Guids.a.p/src_dxguids.cpp.obj
FAILED: subprojects/DirectX-Headers-1.0/libDirectX-Guids.a.p/src_dxguids.cpp.obj 
ccache x86_64-w64-mingw32-g++ -Isubprojects/DirectX-Headers-1.0/libDirectX-Guids.a.p -Isubprojects/DirectX-Headers-1.0 -I../subprojects/DirectX-Headers-1.0 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++14 -g -Wno-error=format -Wno-error=format-extra-args -isystem../subprojects/DirectX-Headers-1.0/include -MD -MQ subprojects/DirectX-Headers-1.0/libDirectX-Guids.a.p/src_dxguids.cpp.obj -MF subprojects/DirectX-Headers-1.0/libDirectX-Guids.a.p/src_dxguids.cpp.obj.d -o subprojects/DirectX-Headers-1.0/libDirectX-Guids.a.p/src_dxguids.cpp.obj -c ../subprojects/DirectX-Headers-1.0/src/dxguids.cpp
In file included from ../subprojects/DirectX-Headers-1.0/src/dxguids.cpp:12:
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21245:13: error: '_In_opt_count_' has not been declared
21245 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |             ^~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21245:28: error: conflicting declaration 'int NumCastableFormats'
21245 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |             ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21244:20: note: previous declaration as 'UINT32 NumCastableFormats'
21244 |             UINT32 NumCastableFormats,
      |             ~~~~~~~^~~~~~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21245:49: error: expected ',' or '...' before 'DXGI_FORMAT'
21245 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |                                                 ^~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21256:13: error: '_In_opt_count_' has not been declared
21256 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |             ^~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21256:28: error: conflicting declaration 'int NumCastableFormats'
21256 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |             ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21255:20: note: previous declaration as 'UINT32 NumCastableFormats'
21255 |             UINT32 NumCastableFormats,
      |             ~~~~~~~^~~~~~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21256:49: error: expected ',' or '...' before 'DXGI_FORMAT'
21256 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |                                                 ^~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21266:13: error: '_In_opt_count_' has not been declared
21266 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |             ^~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21266:28: error: conflicting declaration 'int NumCastableFormats'
21266 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |             ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21265:20: note: previous declaration as 'UINT32 NumCastableFormats'
21265 |             UINT32 NumCastableFormats,
      |             ~~~~~~~^~~~~~~~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:21266:49: error: expected ',' or '...' before 'DXGI_FORMAT'
21266 |             _In_opt_count_(NumCastableFormats)  DXGI_FORMAT *pCastableFormats,
      |                                                 ^~~~~~~~~~~
In file included from ../subprojects/DirectX-Headers-1.0/src/dxguids.cpp:12:
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:22443:5: error: '_In_count_' has not been declared
22443 |     _In_count_(NumFeatures) const IID*     pIIDs,
      |     ^~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:22443:16: error: conflicting declaration 'int NumFeatures'
22443 |     _In_count_(NumFeatures) const IID*     pIIDs,
      |     ~~~~~~~~~~~^~~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:22442:45: note: previous declaration as 'UINT NumFeatures'
22442 |     UINT                                    NumFeatures,
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
../subprojects/DirectX-Headers-1.0/include/directx/d3d12.h:22443:29: error: expected ',' or '...' before 'const'
22443 |     _In_count_(NumFeatures) const IID*     pIIDs,
      |                             ^~~~~
[38/1064] Compiling C object src/util/libmesa_util.a.p/log.c.obj
[39/1064] Compiling C object src/util/libmesa_util.a.p/dag.c.obj
```
The error is at https://gitlab.freedesktop.org/lygstate/mesa/-/jobs/22003018

Maybe it's should be
```
#if !defined(_WIN32) || defined(NTDDI_WIN10_FE) || defined(__MINGW32__)
#include <ocidl.h>
#else
#include <OCIdl.h>
#endif
```

Or using
https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html

Reply:
ping for ack.

Reply:
Can you update the headers to use:

```
#if !defined(_WIN32) || defined(NTDDI_WIN10_FE) || defined(__MINGW32__)
#include <ocidl.h>
#else
#include <OCIdl.h>
#endif
```

Reply:
I still raised a concern, as `include/wsl/stubs/OCIdl.h` are uppercase, so maybe we only need this:
```
#if defined(__MINGW32__)
#include <ocidl.h>
#else
#include <OCIdl.h> /* Linux won't have OCIdl.h, only come  from `include/wsl/stubs/OCIdl.h`  */
#endif
```
For msvc, it's using Windows SDK and won't be a issue.

Reply:
OK, NTDDI_WIN10_FE also guarded